core/src/main/java/bisq/core/app/BisqSetup.java

```diff
@@ -482,14 +482,31 @@ private void maybeShowTac() {
         }
     }
 
-    private void detectLocalBitcoinNode(Runnable nextStep) {
+    private void maybeCheckLocalBitcoinNode(Runnable nextStep) {
         BaseCurrencyNetwork baseCurrencyNetwork = config.baseCurrencyNetwork;
-        if (config.ignoreLocalBtcNode || baseCurrencyNetwork.isDaoRegTest() || baseCurrencyNetwork.isDaoTestNet()) {
+
+        var shouldIgnoreLocalNode =
+            config.ignoreLocalBtcNode
+            || baseCurrencyNetwork.isDaoRegTest()
+            || baseCurrencyNetwork.isDaoTestNet();
+        if (shouldIgnoreLocalNode) {
             nextStep.run();
             return;
         }
 
-        localBitcoinNode.detectAndRun(nextStep);
+        // Results of the check don't have to be passed to nextStep,
+        // because they're cached in LocalBitcoinNode and dependent routines query it themselves.
+        localBitcoinNode.checkUsable();
+
+        // Here we only want to provide the user with a choice (in a popup) in case a local node is
+        // detected, but badly configured.
+        var detectedButMisconfigured = localBitcoinNode.isDetectedButMisconfigured().get();
```

Comment:
I took your suggestion about getting rid of most of the Optional.get()s in other places, but in my opinion here a check would be superfluous, because it's easy to see that getting an empty Optional is impossible here with the current implementation of LocalBitcoinNode (`.checkUsable()` on the previous line guarantees that). That said if it were to change, all the calls would have to be reconsidered anyway, right?